src/libsyntax/parse/parser.rs

```diff
@@ -1661,6 +1661,8 @@ impl<'a> Parser<'a> {
 
     /// matches '-' lit | lit
     pub fn parse_pat_literal_maybe_minus(&mut self) -> PResult<'a, P<Expr>> {
+        maybe_whole_expr!(self);
```

Comment:
These kinds of things are usually checked in `ast_validation` because AST fragments can be produced by procedural macros as well (and parser checks won't help in this case).